lib/lua/Thrift.lua

```diff
@@ -233,6 +235,35 @@ function TException:write(oprot)
   oprot:writeStructEnd()
 end
 
+TUUID = {
+  zero,
+  one,
+  two,
+  three
+}
+
+TUUID = __TObject:new{
+  __type = 'TUUID'
+}
+
+function TUUIDfromString(str)
+  local iterator = string.gmatch(str, "[A-Fa-f0-9][A-Fa-f0-9][A-Fa-f0-9][A-Fa-f0-9]")
+  return TUUID:new {
+    zero = libluabitwise.buor(libluabitwise.ushiftl(tonumber(iterator(), 16), 16), tonumber(iterator(), 16)),
```

Comment:
note that bitshifting was added in Lua 5.3, so if support for Lua 5.2 is dropped this implementation and the entire `libluabitwise` C module can be deleted.